### ISSUE TITLE
Training on a custom dataset

### ISSUE THREAD

**Issue by ys830** [cvlab-columbia/zero123#88]
I appreciate your great work in zero123.

I want to retrain zero123 on medical data. My dataset contains about 700 samples, using the same data processing method as in the paper. Each sample has 12 different perspective images. I trained using a single A100 GPU. Below are my configuration file settings and training logs.

During inference, I used the saved 'last.ckpt' from training, but the generated image perspective control isn't very accurate. There's a deviation in perspective compared to the ground truth images. What could be the reason for this?

Could there be an issue with my configuration settings leading to a higher loss during training convergence?

```
model:
  base_learning_rate: 0.0001
  target: ldm.models.diffusion.ddpm.LatentDiffusion
  params:
    linear_start: 0.00085
    linear_end: 0.012
    num_timesteps_cond: 1
    log_every_t: 200
    timesteps: 1000
    first_stage_key: image_target
    cond_stage_key: image_cond
    image_size: 32
    channels: 4
    cond_stage_trainable: false
    conditioning_key: hybrid
    monitor: val/loss_simple_ema
    scale_factor: 0.18215
    scheduler_config:
      target: ldm.lr_scheduler.LambdaLinearScheduler
      params:
        warm_up_steps:
        - 100
        cycle_lengths:
        - 10000000000000
        f_start:
        - 1.0e-06
        f_max:
        - 1.0
        f_min:
        - 1.0
    unet_config:
      target: ldm.modules.diffusionmodules.openaimodel.UNetModel
      params:
        image_size: 32
        in_channels: 8
        out_channels: 4
        model_channels: 320
        attention_resolutions:
        - 4
        - 2
        - 1
        num_res_blocks: 2
        channel_mult:
        - 1
        - 2
        - 4
        - 4
        num_heads: 8
        use_spatial_transformer: true
        transformer_depth: 1
        context_dim: 768
        use_checkpoint: true
        legacy: false
    first_stage_config:
      target: ldm.models.autoencoder.AutoencoderKL
      params:
        embed_dim: 4
        monitor: val/rec_loss
        ddconfig:
          double_z: true
          z_channels: 4
          resolution: 256
          in_channels: 3
          out_ch: 3
          ch: 128
          ch_mult:
          - 1
          - 2
          - 4
          - 4
          num_res_blocks: 2
          attn_resolutions: []
          dropout: 0.0
        lossconfig:
          target: torch.nn.Identity
    cond_stage_config:
      target: ldm.modules.encoders.modules.FrozenCLIPImageEmbedder
data:
  target: ldm.data.simple.ObjaverseDataModuleFromConfig
  params:
    root_dir: /data/yisi/mywork/zero123/diffusioncardiac
    batch_size: 16
    num_workers: 8
    total_view: 4
    train:
      validation: false
      image_transforms:
        size: 256
    validation:
      validation: true
      image_transforms:
        size: 256
lightning:
  find_unused_parameters: false
  metrics_over_trainsteps_checkpoint: true
  modelcheckpoint:
    params:
      every_n_train_steps: 5000
  callbacks:
    image_logger:
      target: main.ImageLogger
      params:
        batch_frequency: 500
        max_images: 32
        increase_log_steps: false
        log_first_step: true
        log_images_kwargs:
          use_ema_scope: false
          inpaint: false
          plot_progressive_rows: false
          plot_diffusion_rows: false
          'N': 32
          unconditional_guidance_scale: 3.0
          unconditional_guidance_label:
          - ''
  trainer:
    benchmark: true
    val_check_interval: 5000000
    num_sanity_val_steps: 0
    accumulate_grad_batches: 1
    accelerator: ddp
    check_val_every_n_epoch: 10
    gpus: 0,
```
![image](https://github.com/cvlab-columbia/zero123/assets/71709227/3d362ddb-19cd-4c7b-92ee-a1bb107cf1c3)

**Comment by trltzy** [comment]
I have the same problem, how to fix it?

**Comment by perturbedmuffin** [comment]
I am commenting for visibility since I am also having a related issue.

**Comment by ChengWan9** [comment]
can you help me how to prepare my own data to train?what data structure should I process?very thanks!!!

**Comment by ChengWan9** [comment]
> I am commenting for visibility since I am also having a related issue.

can you help me how to prepare my own data to train?what data structure should I process?very thanks!!!

**Comment by kada0720** [comment]
@gitsawww Do you know how to use custom data to train now?

**Comment by BaochaoZhu** [comment]
@ys830 @trltzy @gitsawww @kada0720 Could you please assist me with preparing my data for training? What data structure should I be processing? Thank you.

**Comment by sungmin9939** [comment]
total view should be 12 since you prepared 12 different images for each object

**Comment by kuailexiaohunzi** [comment]
> @ys830 @trltzy @gitsawww @kada0720 Could you please assist me with preparing my data for training? What data structure should I be processing? Thank you.

Have you solved this problem? I also have the same question

**Comment by dzc392** [comment]
> > I am commenting for visibility since I am also having a related issue.我发表评论是为了提高可见性，因为我也遇到了相关问题。
> 
> can you help me how to prepare my own data to train?what data structure should I process?very thanks!!!你能帮我如何准备我自己的数据来训练吗？我应该处理什么数据结构？非常感谢！

i also want to do this . do you know？

**Comment by dzc392** [comment]
> I appreciate your great work in zero123.我很欣赏你在 Zero123 中所做的出色工作。
> 
> I want to retrain zero123 on medical data. My dataset contains about 700 samples, using the same data processing method as in the paper. Each sample has 12 different perspective images. I trained using a single A100 GPU. Below are my configuration file settings and training logs.我想根据医疗数据重新训练 Zero123。我的数据集包含大约 700 个样本，使用与论文中相同的数据处理方法。每个样本有 12 个不同的透视图像。我使用单个 A100 GPU 进行训练。以下是我的配置文件设置和训练日志。
> 
> During inference, I used the saved 'last.ckpt' from training, but the generated image perspective control isn't very accurate. There's a deviation in perspective compared to the ground truth images. What could be the reason for this?在推理过程中，我使用了训练中保存的“last.ckpt”，但生成的图像透视控制不是很准确。与地面真实图像相比，视角存在偏差。这可能是什么原因？
> 
> Could there be an issue with my configuration settings leading to a higher loss during training convergence?我的配置设置是否存在问题，导致训练收敛期间损失更高？
> 
> ```
> model:
>   base_learning_rate: 0.0001
>   target: ldm.models.diffusion.ddpm.LatentDiffusion
>   params:
>     linear_start: 0.00085
>     linear_end: 0.012
>     num_timesteps_cond: 1
>     log_every_t: 200
>     timesteps: 1000
>     first_stage_key: image_target
>     cond_stage_key: image_cond
>     image_size: 32
>     channels: 4
>     cond_stage_trainable: false
>     conditioning_key: hybrid
>     monitor: val/loss_simple_ema
>     scale_factor: 0.18215
>     scheduler_config:
>       target: ldm.lr_scheduler.LambdaLinearScheduler
>       params:
>         warm_up_steps:
>         - 100
>         cycle_lengths:
>         - 10000000000000
>         f_start:
>         - 1.0e-06
>         f_max:
>         - 1.0
>         f_min:
>         - 1.0
>     unet_config:
>       target: ldm.modules.diffusionmodules.openaimodel.UNetModel
>       params:
>         image_size: 32
>         in_channels: 8
>         out_channels: 4
>         model_channels: 320
>         attention_resolutions:
>         - 4
>         - 2
>         - 1
>         num_res_blocks: 2
>         channel_mult:
>         - 1
>         - 2
>         - 4
>         - 4
>         num_heads: 8
>         use_spatial_transformer: true
>         transformer_depth: 1
>         context_dim: 768
>         use_checkpoint: true
>         legacy: false
>     first_stage_config:
>       target: ldm.models.autoencoder.AutoencoderKL
>       params:
>         embed_dim: 4
>         monitor: val/rec_loss
>         ddconfig:
>           double_z: true
>           z_channels: 4
>           resolution: 256
>           in_channels: 3
>           out_ch: 3
>           ch: 128
>           ch_mult:
>           - 1
>           - 2
>           - 4
>           - 4
>           num_res_blocks: 2
>           attn_resolutions: []
>           dropout: 0.0
>         lossconfig:
>           target: torch.nn.Identity
>     cond_stage_config:
>       target: ldm.modules.encoders.modules.FrozenCLIPImageEmbedder
> data:
>   target: ldm.data.simple.ObjaverseDataModuleFromConfig
>   params:
>     root_dir: /data/yisi/mywork/zero123/diffusioncardiac
>     batch_size: 16
>     num_workers: 8
>     total_view: 4
>     train:
>       validation: false
>       image_transforms:
>         size: 256
>     validation:
>       validation: true
>       image_transforms:
>         size: 256
> lightning:
>   find_unused_parameters: false
>   metrics_over_trainsteps_checkpoint: true
>   modelcheckpoint:
>     params:
>       every_n_train_steps: 5000
>   callbacks:
>     image_logger:
>       target: main.ImageLogger
>       params:
>         batch_frequency: 500
>         max_images: 32
>         increase_log_steps: false
>         log_first_step: true
>         log_images_kwargs:
>           use_ema_scope: false
>           inpaint: false
>           plot_progressive_rows: false
>           plot_diffusion_rows: false
>           'N': 32
>           unconditional_guidance_scale: 3.0
>           unconditional_guidance_label:
>           - ''
>   trainer:
>     benchmark: true
>     val_check_interval: 5000000
>     num_sanity_val_steps: 0
>     accumulate_grad_batches: 1
>     accelerator: ddp
>     check_val_every_n_epoch: 10
>     gpus: 0,
> ```
> 
> ![image](https://private-user-images.githubusercontent.com/71709227/264521495-3d362ddb-19cd-4c7b-92ee-a1bb107cf1c3.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjk0MjYwMTAsIm5iZiI6MTcyOTQyNTcxMCwicGF0aCI6Ii83MTcwOTIyNy8yNjQ1MjE0OTUtM2QzNjJkZGItMTljZC00YzdiLTkyZWUtYTFiYjEwN2NmMWMzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDIwVDEyMDE1MFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWYxNTJmZDE0MzM2NjFjNGQyNDZiYmNiMzFiOWMyZTA3NzlkODRjZTNlYWE0NjhjNjFjY2NjMGZjMzI0OWZjNzYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.hau5-CXmTJcIa8WCgG72yHPk92wk_7vDCx0_WlCp5w4)

Could you tell me how to render  from 3D,thanks!
